google/cloud/security/scanner/audit/iap_rules_engine.py

```diff
@@ -0,0 +1,261 @@
+
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Rules engine for IAP policies on backend services"""
+from collections import namedtuple
+import itertools
+import re
+
+from google.cloud.security.common.util import log_util
+from google.cloud.security.scanner.audit import base_rules_engine as bre
+from google.cloud.security.scanner.audit import errors as audit_errors
+
+LOGGER = log_util.get_logger(__name__)
+
+
+# TODO: move this to utils since it's used in more that one engine
+def escape_and_globify(pattern_string):
```

Comment:
Just a note. In PR #365 I moved this one in utils, since it's used so much. Depending when this and #365 get merged, but maybe the TODO should say that the next author changes that